sdk/tests/conformance2/reading/read-pixels-into-pixel-pack-buffer.html

```diff
@@ -0,0 +1,269 @@
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Type should be able to read as " + wtu.glEnumToString(gl, type));
+    }
+
+    var invalidFormat = [gl.DEPTH_COMPONENT, gl.DEPTH_STENCIL, gl.R8, gl.RGBA4];
+    var invalidType = [gl.UNSIGNED_INT_24_8];
+    for (var ff = 0; ff < invalidFormat.length; ++ff) {
+        var format = invalidFormat[ff];
+        gl.readPixels(0, 0, 1, 1, format, gl.UNSIGNED_BYTE, 0);
+        wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "Format should not be able to read as " + wtu.glEnumToString(gl, format));
+    }
+    for (var tt = 0; tt < invalidType.length; ++tt) {
+        var type = invalidType[tt];
+        gl.readPixels(0, 0, 1, 1, gl.RGBA, type, 0);
+        wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "Type should not be able to read as " + wtu.glEnumToString(gl, type));
+    }
+
+    debug("");
+    debug("check combinations of format and type");
+    var combinations = [
+        {format: gl.RGBA,         type: gl.UNSIGNED_BYTE},
+        {format: gl.RGB,          type: gl.UNSIGNED_BYTE},
+        {format: gl.RGB,          type: gl.UNSIGNED_SHORT_5_6_5},
+        {format: gl.RGBA,         type: gl.UNSIGNED_SHORT_5_5_5_1},
+        {format: gl.RGBA,         type: gl.UNSIGNED_SHORT_4_4_4_4},
+        {format: gl.ALPHA,        type: gl.UNSIGNED_BYTE},
+        {format: gl.LUMINANCE,    type: gl.UNSIGNED_BYTE},
+        {format: gl.RED,          type: gl.UNSIGNED_BYTE},
+        {format: gl.RGBA_INTEGER, type: gl.UNSIGNED_INT},
+        {format: gl.RGBA_INTEGER, type: gl.INT}
+    ];
+
+    var implFormat = gl.getParameter(gl.IMPLEMENTATION_COLOR_READ_FORMAT);
+    var implType = gl.getParameter(gl.IMPLEMENTATION_COLOR_READ_TYPE);
+    for (var tt = 0; tt < combinations.length; ++ tt) {
+        var info = combinations[tt];
+        var format = info.format;
+        var type = info.type;
+        gl.readPixels(0, 0, 1, 1, format, type, 0);
+        // Only two format/type parameter pairs are accepted. GL_RGBA/GL_UNSIGNED_BYTE is always
+        // accepted on default readbuffer. The other acceptable pair can be discovered by querying
+        // GL_IMPLEMENTATION_COLOR_READ_FORMAT and GL_IMPLEMENTATION_COLOR_READ_TYPE.
+        if ((format == gl.RGBA && type == gl.UNSIGNED_BYTE) || (format == implFormat && type == implType)) {
+            wtu.glErrorShouldBe(gl, gl.NO_ERROR, "The combination of format/type should be able to read as " +
+                                wtu.glEnumToString(gl, format) + " / " + wtu.glEnumToString(gl, type));
+        } else {
+            wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "The combination of format/type should not be able to read as " +
+                                wtu.glEnumToString(gl, format) + " / " + wtu.glEnumToString(gl, type));
+        }
+    }
+}
+
+function validatePixelPackBufferAndParameters()
+{
+    debug("");
+    debug("Validate PIXEL_PACK buffer and readPixels' parameters");
+    var canvasWidth = 2, canvasHeight = 2;
```

Comment:
It's better for width/height to be passed in as function parameters, to be consistent with other functions in this file.